File: Models/DsngayDocSoBilling.cs
using System;
using System.Collections.Generic;

namespace WebAPI_NRW.Models.Database;

public partial class DsngayDocSoBilling
{
    public int Id { get; set; }

    public int Nam { get; set; }

    public int Ky { get; set; }

    public int SoNgayDocSoBilling { get; set; }

    public string? GhiChu { get; set; }

    public DateTime NgayTao { get; set; }

    public DateTime? NgayCapNhat { get; set; }

    public int NguoiTao { get; set; }

    public int? NguoiCapNhat { get; set; }

    public virtual ICollection<DsngayDocSoBillingChiTiet> DsngayDocSoBillingChiTiets { get; set; } = new List<DsngayDocSoBillingChiTiet>();

    public virtual NguoiDung? NguoiCapNhatNavigation { get; set; }

    public virtual NguoiDung NguoiTaoNavigation { get; set; } = null!;
}

File: src/types/he-thong-billing/ds-ngay-doc-so-billing.ts
/* ========== REQUEST (CLIENT --> API) ========== */
// Thêm dữ liệu
export interface AddDsNgayDocSoBillingRequest {
  nam: number;
  ky: number;
  soNgayDocSoBilling: number;
  ghiChu?: string;
  ngayTao: string;
  nguoiTao: number;
}

// Cập nhật dữ liệu
export interface UpdateDsNgayDocSoBillingRequest {
  nam: number;
  ky: number;
  soNgayDocSoBilling: number;
  ghiChu?: string;
  ngayCapNhat?: string;
  nguoiCapNhat?: number;
}

/* ========== RESPONSE (API --> CLIENT) ========== */
export interface DsNgayDocSoBillingResponse {
  id: number;
  nam: number;
  ky: number;
  soNgayDocSoBilling: number;
  ghiChu?: string;
  ngayTao: string;
  ngayCapNhat?: string;
  nguoiTao?: string;
  nguoiCapNhat?: string;
}
